shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/variantcallers/HaplotypeCallerGvcf.scala

```
@@ -34,14 +37,82 @@ class HaplotypeCallerGvcf(val parent: Configurable) extends Variantcaller {
+      case (Some(l), None) => l.length.toDouble / refernceSize
+      case (None, Some(l)) => l.length.toDouble / refernceSize
+      case (Some(l1), Some(l2)) =>
+        BedRecordList.fromList(l1.allRecords ++ l2.allRecords).combineOverlap.length.toDouble / refernceSize
+      case (None, None) => 0.0
+    }
+  }
+
   override def fixedValues = Map("haplotypecaller" -> Map("emitRefConfidence" -> "GVCF"))
 
+  override def defaults = Map("haplotypecaller" -> Map(
+    "variant_index_type" -> "LINEAR",
+    "variant_index_parameter" -> 128000)
+  )
+
   def biopetScript() {
     gVcfFiles = for ((sample, inputBam) <- inputBams) yield {
-      val hc = gatk.HaplotypeCaller(this, List(inputBam), new File(outputDir, sample + ".gvcf.vcf.gz"))
-      hc.BQSR = inputBqsrFiles.get(sample)
-      add(hc)
-      sample -> hc.out
+      if (genderAwareCalling) {
+        val finalFile = new File(outputDir, sample + ".gvcf.vcf.gz")
+        val gender = genders.getOrElse(sample, Gender.Unknown)
+        val haploidBedFiles: List[File] = gender match {
+          case Gender.Female => haploidRegions.toList ::: haploidRegionsFemale.toList ::: Nil
+          case Gender.Male   => haploidRegions.toList ::: haploidRegionsMale.toList ::: Nil
+          case _             => haploidRegions.toList
+        }
+
+        val fraction: Double = gender match {
+          case Gender.Female => fractionFemale
+          case Gender.Male   => fractionMale
+          case _             => fractionUnknown
+        }
+
+        val haploidGvcf = if (haploidBedFiles.nonEmpty) {
+          val hc = gatk.HaplotypeCaller(this, List(inputBam), new File(outputDir, sample + ".haploid.gvcf.vcf.gz"))
+          hc.BQSR = inputBqsrFiles.get(sample)
+          hc.intervals = haploidBedFiles
+          hc.scatterCount = (hc.scatterCount * fraction).toInt
+          hc.sample_ploidy = Some(1)
+          add(hc)
+          Some(hc.out)
+        } else None
```

It's probably better to give warning/error that gender-aware calling does not make sense/is impossible when there are no haploid bed files, in stead of just silently skipping the step.

---

This would mean that if you not giving any haploid bed file for the general one and females you get this message for each female sample. Still this warning should be given but at an other place. Will add something for this.